estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,63 @@
+from dateutil.relativedelta import relativedelta
+from odoo import api, models, fields
+from odoo.exceptions import UserError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "estate property offer"
+    _order = "price desc"
+
+    price = fields.Float()
+    status = fields.Selection(
+        selection=[('accepted', 'Accepted'), ('refused', 'Refused')],
+        copy=False
+    )
+    partner_id = fields.Many2one('res.partner', string="Partner", required=True)
+    property_id = fields.Many2one('estate.property', string="Property", required=True, ondelete="cascade")
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(compute="_compute_date_deadline", inverse="_inverse_date_deadline")
+    property_state = fields.Selection(related="property_id.state")
+    property_type_id = fields.Many2one(related="property_id.property_type_id", store=True)
+
+    _positive_offer_price = models.Constraint(
+        'CHECK(price > 0)',
+        'The Offer Price must be strictly positive'
+    )
+
+    @api.depends("validity", "create_date")
+    def _compute_date_deadline(self):
+        for offer in self:
+            offer.date_deadline = (offer.create_date or fields.Date.today()) + relativedelta(days=+offer.validity)
+
+    def _inverse_date_deadline(self):
+        for offer in self:
+            offer.validity = (offer.date_deadline - (offer.create_date or fields.Date.today()).date()).days
+
+    @api.model
+    def create(self, vals):
+        for offer in vals:
+            prop = self.env['estate.property'].browse(offer['property_id'])
+            prop.state = "offer_received"
+            if offer['price'] < prop.best_price:
+                raise UserError(f"The offer must be higher than {prop.best_price}")
+        return super().create(vals)
+
+    def accept_offer(self):
```

Comment:
As this method is used by a button, its name must start with "action". You can find the Odoo guidelines right [here](https://www.odoo.com/documentation/19.0/contributing/development/coding_guidelines.html?) 😃.
```suggestion
    def action_accept_offer(self):
```